### PR TITLE
Add/database set up

### PR DESCRIPTION
**Pull request by damienlethiec**
- add main models and relations
- add bullet and annotate gem

Questions:
1) Should we create a separate Stage table with the following relations: a festival has many stages and a stage as many concerts. It would make our stages more consistent than just storing them as string in concerts and would allow us to add more info on them in the future. If yes, we should remove the 1:N relation between concerts and a festival (a festival has many concerts through stages)
2) Same question for the Genre, should we add a Genre table? If we want to be precise, we would need a join table between Genre and Artist. Same thing, in the future it could allow us to add more infos easily
2) What do you think of the Invite model like in this tutorial: https://coderwall.com/p/rqjjca/creating-a-scoped-invitation-system-for-rails. Personally I think it is great.

I did not add all the omniauth column is User. We will create them later.

In the end, if we adopt of the points in my questions, the db would like something like this at the beginning
image

### CODE REVIEW

File: db/migrate/20170309215132_create_festivals.rb [hunk 1]
```diff
@@ -0,0 +1,13 @@
+class CreateFestivals < ActiveRecord::Migration[5.0]
+  def change
+    create_table :festivals do |t|
+      t.string :name
+      t.time :start_date
+      t.time :end_date
+      t.string :location
+      t.string :photo
```
Comment (Jeanquetil) [inline db/migrate/20170309215132_create_festivals.rb]: On veut un champ photo pour les festivals ? Pour l'instant je n'en vois pas l'intérêt perso

File: db/migrate/20170309215132_create_festivals.rb [hunk 2]
```diff
@@ -0,0 +1,13 @@
+class CreateFestivals < ActiveRecord::Migration[5.0]
+  def change
+    create_table :festivals do |t|
+      t.string :name
+      t.time :start_date
+      t.time :end_date
+      t.string :location
```
Comment (Jeanquetil) [inline db/migrate/20170309215132_create_festivals.rb]: La location ce sera quoi ? Une ville ? Une adresse ? Un pays ? Je serais plus explicite sur le nom de cette colonne

File: db/migrate/20170309220508_add_attributes_to_user.rb [hunk 3]
```diff
@@ -0,0 +1,7 @@
+class AddAttributesToUser < ActiveRecord::Migration[5.0]
+  def change
+    add_column :users, :first_name, :string
+    add_column :users, :last_name, :string
+    add_column :users, :admin, :boolean
```
Comment (Jeanquetil) [inline db/migrate/20170309220508_add_attributes_to_user.rb]: nitpicking mais j'appellerais la colonne `is_admin` plutôt pour une meilleure lisibilité :-)

`user.admin` <=> je ne m'attends pas forcément à ce que ça retourne un booléen
`user.is_admin` <=> ça me suggère forcément un booléen

**Comment by damienlethiec** [comment]
Ok pour les genres et les stages @Jeanquetil , ça me va bien ! Je rajoute le modèle invite, je fais les petites modifs demandée et je repush en te pinguant si tu veux merger ;)

**Pull request merged**